src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -304,12 +304,19 @@ def _build_os_profile():
         if enable_auto_update is not None and custom_image_os_type.lower() == 'windows':
             os_profile['windowsConfiguration']['enableAutomaticUpdates'] = enable_auto_update
 
+        # Windows patch settings
         if patch_mode is not None and custom_image_os_type.lower() == 'windows':
             os_profile['windowsConfiguration']['patchSettings'] = {
                 'patchMode': patch_mode,
                 'enableHotpatching': enable_hotpatching
             }
 
+        # Linux patch settings
+        if linux_patch_mode is not None and custom_image_os_type.lower() == 'linux':
```

Comment:
what if user not using custom_image? None check

Reply:
If can't infer OS type, user must provide one. OS type is inferred in validator.